### PR TITLE
Document usage of Apollo REST link

### PR DESCRIPTION
**Pull request by emmenko**
Follow up of #2733

### CODE REVIEW

File: website/src/content/api-reference/commercetools-frontend-application-shell.mdx [hunk 1]
```diff
@@ -113,17 +113,33 @@ const EntryPoint = () => {
...
+
+* `restLink` (optional): Instance of the [Apollo REST link](https://www.apollographql.com/docs/react/api/link/apollo-link-rest/).
+
+  <Info>
+
+  This feature is available from version `21.10.0` onwards.
+
+  The `apollo-link-rest` and its related dependencies are not included in the `@commercetools-frontend/application-shell` package and must be installed separately.
+
+  When configuring the REST link, we recommend setting the `uri` using the `getMcApiUrl()` utility function.
```
Comment (emmenko) [inline website/src/content/api-reference/commercetools-frontend-application-shell.mdx:138]: Requires #2723 to be released first.

File: website/src/content/development/data-fetching.mdx [hunk 2]
```diff
@@ -532,6 +532,126 @@ const HelloWorld = () => {
...
+}
+```
+
+A few things to note:
+* Types and nested types require to define the `__typename` via `@rest(type: "")` or `@type(name: "")`.
+* The `endpoint` is passed as a variable and should point to the [Merchant Center API proxy target endpoint](/concepts/merchant-center-api#proxytarget).
+* For `LocalizedString` fields such as `name` (in the REST API) you should use the GraphQL naming convention `nameAllLocales`. However, you also need to instruct the Apollo REST link to correctly transform the shape of the field. You can do that using the [responseTransformer](https://www.apollographql.com/docs/react/api/link/apollo-link-rest/#response-transforming) function.
+
+  ```js
+  import { applyTransformedLocalizedStrings } from '@commercetools-frontend/l10n';
```
Comment (emmenko) [inline website/src/content/development/data-fetching.mdx:627]: Requires #2749 to be released first

**Comment by emmenko** [comment]
This is ready for review now.

File: website/src/content/api-reference/commercetools-frontend-application-shell.mdx (hunk 1 above)
Comment (CarlosCortizasCT) [inline website/src/content/api-reference/commercetools-frontend-application-shell.mdx:138]: nit: perhaps we could explain this utility function is also exported from this package.

**Review by CarlosCortizasCT — APPROVED** [review]
💯

**Review by kark — APPROVED** [review]

**Review by frontastic-developers — APPROVED** [review]
Looks good to me.

**Review by Anshuman71 — APPROVED** [review]
Looks good to me!

**Comment by Anshuman71** [comment]
> Looks good to me.

Sorry, I used the team account earlier.

**Review by ddouglasz — APPROVED** [review]
Looks good 👏

**Pull request merged**